tests/bsim/bluetooth/host/iso/frag_2/src/broadcaster.c

```diff
@@ -0,0 +1,277 @@
+	 * Open issue: https://github.com/zephyrproject-rtos/zephyr/issues/72852
+	 */
+	k_sleep(K_MSEC(100));
+
+	/* Enable Periodic Advertising */
+	err = bt_le_per_adv_start(adv);
+	TEST_ASSERT(err == 0, "Failed to enable periodic advertising: %d", err);
+}
+
+static void create_big(struct bt_le_ext_adv *adv, size_t cnt, struct bt_iso_big **out_big)
+{
+	const uint16_t latency_ms = 10U;
+	const uint16_t sdu_interval_us = 10U * USEC_PER_MSEC;
+
+	struct bt_iso_chan *channels[ARRAY_SIZE(iso_chans)];
+	struct bt_iso_big_create_param param = {
+		.packing = BT_ISO_PACKING_SEQUENTIAL,
+		.framing = BT_ISO_FRAMING_UNFRAMED,
+		.interval = sdu_interval_us,
+		.bis_channels = channels,
+		.latency = latency_ms,
+		.encryption = false,
+		.num_bis = cnt,
+	};
+	int err;
+
+	for (size_t i = 0U; i < cnt; i++) {
+		channels[i] = &iso_chans[i];
+	}
+
+	LOG_INF("Creating BIG");
+
+	err = bt_iso_big_create(adv, &param, out_big);
+	TEST_ASSERT(err == 0, "Failed to create BIG: %d", err);
+
+	WAIT_FOR_FLAG(iso_connected);
+}
+
+struct bt_le_ext_adv *adv;
+struct bt_iso_big *big;
+
+static struct bt_iso_chan_ops iso_ops = {
+	.disconnected = iso_disconnected_cb,
+	.connected = iso_connected_cb,
+	.sent = sdu_sent_cb,
+};
+static struct bt_iso_chan_io_qos iso_tx = {
+	.sdu = CONFIG_BT_ISO_TX_MTU,
+	.phy = BT_GAP_LE_PHY_2M,
+	.rtn = 1,
+	.path = NULL,
+};
+static struct bt_iso_chan_qos iso_qos = {
+	.tx = &iso_tx,
+	.rx = NULL,
+};
+
+static void init(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	TEST_ASSERT(err == 0, "Bluetooth enable failed: %d", err);
+}
+
+static void connect_iso(void)
+{
+	bt_testing_set_iso_mtu(10);
+
+	for (size_t i = 0U; i < ARRAY_SIZE(iso_chans); i++) {
+		iso_chans[i].ops = &iso_ops;
+		iso_chans[i].qos = &iso_qos;
+	}
+
+	create_ext_adv(&adv);
+	create_big(adv, 1U, &big);
+	start_ext_adv(adv);
+}
+
+static void disconnect_iso(void)
+{
+	int err;
+
+	err = bt_iso_big_terminate(big);
+	TEST_ASSERT(err == 0, "bt_iso_big_terminate failed (%d)", err);
+	err = bt_le_per_adv_stop(adv);
+	TEST_ASSERT(err == 0, "bt_le_per_adv_stop failed (%d)", err);
+	k_msleep(100);
+	err = bt_le_ext_adv_stop(adv);
+	TEST_ASSERT(err == 0, "bt_le_ext_adv_stop failed (%d)", err);
+	k_msleep(100);
+	err = bt_le_ext_adv_delete(adv);
+	TEST_ASSERT(err == 0, "bt_le_ext_adv_delete failed (%d)", err);
+
+	big = NULL;
+	adv = NULL;
+}
+
+void entrypoint_broadcaster(void)
+{
+	/* Test purpose:
+	 *
+	 * Verifies that we are not leaking buffers when getting disconnected
+	 * while sending fragmented ISO SDU.
+	 *
+	 * One device:
+	 * - `broadcaster`: sends fragmented ISO SDUs
+	 *
+	 * Procedure:
+	 * - initialize Bluetooth and a BIS
+	 * - send a fragmented SDU
+	 * - disconnect when the first fragment is sent
+	 * - repeat NUM_BUF time
+	 *
+	 * [verdict]
+	 * - no buffer is leaked and repeating the operation success
+	 */
+	int err;
+	uint8_t NUM_BUF = 4;
+
+	LOG_INF("Starting ISO HCI fragmentation test 2");
+
+	init();
+
+	for (int l = 0; l < NUM_BUF; l++) {
```

Comment:
```suggestion
	for (int i = 0; i < NUM_BUF; i++) {
```

Why are you using `l` like some kind of madman :D

Reply:
l == loop
i == iteration
dolores: `Doesn't seem like anything to me`

Reply:
i == iterator
l == looper? :)

Come on - let's just stick to the what's common and not trigger extra cycles in people's brains when they read the code.